### PR TITLE
Add linkerd deployment alert

### PR DESCRIPTION
**Pull request by ubergesundheit**
Towards: https://github.com/giantswarm/giantswarm/issues/21041

This PR adds alerting on linkerd deployment failures

### Checklist

- [x] Update CHANGELOG.md
- [x] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [x] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [x] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

**Review by mcharriere — APPROVED** [review]

File: test/tests/providers/global/linkerd.deployment.rules.test.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+rule_files:
+  - linkerd.deployment.rules.yml
+
+tests:
+  - interval: 1m
+    input_series:
+      - series: 'managed_app_deployment_status_replicas_unavailable{deployment="linkerd-destination", managed_app="destination", namespace="linkerd"}'
+        values: '_x5 0+0x10 1+0x20'
```
Comment (mcharriere) [inline test/tests/providers/global/linkerd.deployment.rules.test.yml:9]: I think your series is 10 minutes short

File: helm/prometheus-rules/templates/alerting-rules/linkerd.deployment.rules.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+spec:
+  groups:
+  - name: linkerd-deployment
+    rules:
+    - alert: LinkerdDeploymentNotSatisfied
+      annotations:
+        description: '{{`Linkerd Deployment {{ $labels.namespace}}/{{ $labels.deployment }} is not satisfied.`}}'
+        opsrecipe: managed-app-linkerd/
+      expr: managed_app_deployment_status_replicas_unavailable{deployment=~"linkerd.*"} > 0
+      for: 30m
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/linkerd.deployment.rules.yml:18]: This means the alert has to match for 30min before it fires.
The data starts erroring at T+15
The unit test expects it to fire at T+25

Does not add up.

File: test/tests/providers/global/linkerd.deployment.rules.test.yml (hunk 1 above)
Comment (ubergesundheit) [reply test/tests/providers/global/linkerd.deployment.rules.test.yml]: Thanks, pushed [4d6cd82](https://github.com/giantswarm/prometheus-rules/pull/689/commits/4d6cd82633d0958f76cb5d6b05730d4caacaf8e3) before reading this

File: helm/prometheus-rules/templates/alerting-rules/linkerd.deployment.rules.yml (hunk 2 above)
Comment (ubergesundheit) [reply helm/prometheus-rules/templates/alerting-rules/linkerd.deployment.rules.yml:18]: Thanks, pushed [4d6cd82](https://github.com/giantswarm/prometheus-rules/pull/689/commits/4d6cd82633d0958f76cb5d6b05730d4caacaf8e3) before reading this

**Pull request merged**